packages/components/src/Popover/Popover.tsx

```diff
@@ -1,146 +1,86 @@
-    popperElement,
-    {
-      modifiers: buildModifiers(arrowElement),
-      placement: preferredPlacement,
-    },
-  );
-  useRefocusOnActivator(open);
-
-  const popoverClassNames = classnames(
-    classes.popover,
-    UNSAFE_className.container,
-  );
-  const dismissButtonClassNames = classnames(
-    classes.dismissButton,
-    UNSAFE_className.dismissButtonContainer,
-  );
-  const arrowClassNames = classnames(classes.arrow, UNSAFE_className.arrow);
-  const popoverContent = (
-    <div
-      role="dialog"
-      data-elevation={"elevated"}
-      ref={setPopperElement}
-      style={{ ...popperStyles.popper, ...(UNSAFE_style.container ?? {}) }}
-      className={popoverClassNames}
-      {...attributes.popper}
-      data-testid="popover-container"
+  return (
+    <Popover.Provider
+      attachTo={attachTo}
+      open={open}
+      preferredPlacement={preferredPlacement}
+      UNSAFE_className={UNSAFE_className}
+      UNSAFE_style={UNSAFE_style}
     >
-      <div
-        className={dismissButtonClassNames}
-        style={UNSAFE_style.dismissButtonContainer ?? {}}
-        data-testid="popover-dismiss-button-container"
-      >
-        <ButtonDismiss onClick={onRequestClose} ariaLabel="Close dialog" />
-      </div>
+      <Popover.DismissButton
+        UNSAFE_className={UNSAFE_className}
+        UNSAFE_style={UNSAFE_style}
+        onClick={onRequestClose}
+      />
       {children}
-      <div
-        ref={setArrowElement}
-        className={arrowClassNames}
-        style={{ ...popperStyles.arrow, ...(UNSAFE_style.arrow ?? {}) }}
-        data-testid="popover-arrow"
+      <Popover.Arrow
+        UNSAFE_className={UNSAFE_className}
+        UNSAFE_style={UNSAFE_style}
       />
-    </div>
+    </Popover.Provider>
   );
-
-  return <>{open && ReactDOM.createPortal(popoverContent, document.body)}</>;
 }
 
-function buildModifiers(arrowElement: HTMLElement | undefined | null) {
-  const modifiers = [
-    {
-      name: "arrow",
-      options: { element: arrowElement, padding: 10 },
-    },
-    {
-      name: "offset",
-      options: {
-        offset: [0, 10],
-      },
-    },
-    {
-      name: "flip",
-      options: {
-        fallbackPlacements: ["auto"],
-      },
-    },
-  ];
+Popover.Provider = PopoverProvider;
 
-  return modifiers;
-}
+Popover.Arrow = function PopoverArrow({
+  UNSAFE_className,
+  UNSAFE_style,
+}: PopoverArrowProps) {
+  const { setArrowElement, popperStyles } = usePopoverContext();
+  const popoverStyles = usePopoverStyles();
+  const classes = classnames(popoverStyles.arrow, UNSAFE_className?.arrow);
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function isHTMLElement(el: any): el is Element {
-  return globalThis?.document && el instanceof Element;
-}
+  return (
+    <div
+      ref={setArrowElement}
+      className={classes}
+      style={{ ...popperStyles.arrow, ...UNSAFE_style?.arrow }}
+      data-testid="popover-arrow"
```

Comment:
I know it's not from this PR but we should try to use `ATL-` for our testids to avoid collision with fairly common names, and also hint at the fact that a test might be dipping its toes into testing Atlantis behavior if it's targeting these containers rather than the expected content

https://jobber.atlassian.net/wiki/spaces/teamuxf/pages/2476933462/How+should+we+declare+test+ID+s+in+Atlantis+components

I do see a couple instances in product where we've already targeted this value, so it might be more effort than we'd like to do right now on that topic. don't need to address it now.